tests/integration/conftest.py

```diff
@@ -389,3 +389,73 @@ def _load_json_schema(path):
 
     with open(path, mode="r") as handler:
         return json.load(handler)
+
+
+@pytest.fixture
+def pre_registered(shell):
+    """
+    A fixture to install subscription manager and pre-register the system prior to the convert2rhel run.
+    """
+    assert shell("yum install -y subscription-manager").returncode == 0
+    # Download the SSL certificate
+    shell("curl --create-dirs -o /etc/rhsm/ca/redhat-uep.pem https://ftp.redhat.com/redhat/convert2rhel/redhat-uep.pem")
+    # Register the system
+    assert (
+        shell(
+            "subscription-manager register --serverurl {} --username {} --password {}".format(
+                env.str("RHSM_SERVER_URL"), env.str("RHSM_USERNAME"), env.str("RHSM_PASSWORD")
+            )
+        ).returncode
+        == 0
+    )
+
+    assert shell("subscription-manager attach --pool {}".format(env.str("RHSM_POOL"))).returncode == 0
+
+    rhsm_uuid_command = "subscription-manager identity | grep identity"
+
+    uuid_raw_output = shell(rhsm_uuid_command).output
+
+    # The `subscription-manager identity | grep identity` command returns
+    # system identity: <UUID>, we need to store just the system UUID
+    original_registration_uuid = uuid_raw_output.split(":")[1].strip()
+
+    yield
+
+    # For some scenarios we do not pre-register the system, therefore we do not have the original UUID
+    # and do not need to verify it stays the same
+    if "C2R_TESTS_CHECK_RHSM_UUID_MATCH" in os.environ:
+        # Get the registered system UUID
+        uuid_raw_output = shell(rhsm_uuid_command).output
+        post_c2r_registration_uuid = uuid_raw_output.split(":")[1].strip()
+
+        # Validate it matches with UUID prior to the conversion
+        assert original_registration_uuid == post_c2r_registration_uuid
+        del os.environ["C2R_TESTS_CHECK_RHSM_UUID_MATCH"]
```

Comment:
I do not like the fact that we are deleting the environment variable in a function (fixture) that was set outside this function. Due to time limitations, I would keep it as it is, but in the future, I would prefer that we parametrize our fixtures so we do not rely on environment variables. This way we would get rid of all these and just stick with regular parameters of the function. This could be done using the indirect parametrization (https://docs.pytest.org/en/7.1.x/example/parametrize.html#indirect-parametrization)

Reply:
Created ticket: https://issues.redhat.com/browse/RHELC-1134

Reply:
I've created the issue @kokesak https://issues.redhat.com/browse/RHELC-1135